### ISSUE TITLE
False positive for SlotReused when having two slots with the same name but different receivers

### ISSUE THREAD

**Issue by francescocervone** [slackhq/compose-lints#434]
With the example below, the rule reports the `content` composable parameter of the `Sample` function as it was called twice. 
Indeed, by renaming either the `Sample` function parameter or the `Section` property, the rule doesn't throw an error anymore.

This scenario might be useful when you need an optional slot and other data exclusively dependent on the existence of the slot itself (what I marked with `// other stuff` in the example).

Example:
```kotlin
data class Section(
  // other stuff
  val content: @Composable () -> Unit,
)
@Composable
fun Sample(
  content: @Composable () -> Unit,
  section: Section?
) {
  Column {
    content()
    if (section != null) {
      // using other stuff
      section.content()
    }
  }
}
```
